eclipse-collections/src/main/java/org/eclipse/collections/impl/lazy/AbstractLazyIterable.java

```diff
@@ -297,4 +302,28 @@ public LazyShortIterable collectShort(ShortFunction<? super T> shortFunction)
         this.forEach(new NonMutatingAggregationProcedure<T, K, V>(map, groupBy, zeroValueFactory, nonMutatingAggregator));
         return map;
     }
+
+    public <V> MutableObjectLongMap<V> sumByInt(Function<? super T, ? extends V> groupBy, IntFunction<? super T> function)
```

Comment:
I'm not sure about the changes to LazyIterables. Their return type is now more specific and is mutable. Is there a reason to prefer mutable here over read-only? LazyIterable is closely tied to RichIterable.